src/util/serialize.cpp

```diff
@@ -354,6 +354,55 @@ std::string deSerializeJsonString(std::istream &is)
 	return os.str();
 }
 
+std::string serializeJsonStringIfNeeded(const std::string &s)
```

Comment:
please stopping returning string, use std::string * as a parameter for your data write